File: example/java/jimmer-sql2/src/main/java/org/babyfish/jimmer/sql2/example/dal/TreeNodeRepository.java
package org.babyfish.jimmer.sql2.example.dal;

import org.babyfish.jimmer.spring.repository.JRepository;
import org.babyfish.jimmer.sql2.example.model.TreeNode;
import org.babyfish.jimmer.sql.fetcher.Fetcher;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public interface TreeNodeRepository extends JRepository<TreeNode, Long> {

    List<TreeNode> findByParentIsNullAndName(
            @Nullable String name,
            Fetcher<TreeNode> fetcher
    );
}

File: example/kotlin/jimmer-sql-kt/src/main/kotlin/org/babyfish/jimmer/example/kt/sql/dal/TreeNodeRepository.kt
package org.babyfish.jimmer.example.kt.sql.dal

import org.babyfish.jimmer.example.kt.sql.model.TreeNode
import org.babyfish.jimmer.spring.repository.KRepository
import org.babyfish.jimmer.sql.fetcher.Fetcher

interface TreeNodeRepository : KRepository<TreeNode, Long> {

    fun findByParentIsNullAndName(
        name: String?,
        fetcher: Fetcher<TreeNode>?
    ): List<TreeNode>
}